test/e2e/tests/single_zone_e2e_test.go

```diff
@@ -81,7 +84,69 @@ var _ = Describe("GCE PD CSI Driver", func() {
 		// Attach Disk
 		err := testAttachWriteReadDetach(volID, volName, instance, client, false /* readOnly */)
 		Expect(err).To(BeNil(), "Failed to go through volume lifecycle")
+	})
+
+	It("Should automatically fix symlink errors between /dev/sdx and /dev/by-id if disk is not found or incorrect", func() {
+		testContext := getRandomTestContext()
+
+		p, z, _ := testContext.Instance.GetIdentity()
+		client := testContext.Client
+		instance := testContext.Instance
+
+		// Set-up instance to have scsi_id where we expect it
+		err := testutils.MkdirAll(instance, "/lib/udev_containerized")
+		Expect(err).To(BeNil(), "could not make scsi_id containerized directory")
+		err = testutils.CopyFile(instance, "/lib/udev/scsi_id", "/lib/udev_containerized/scsi_id")
+		Expect(err).To(BeNil(), "could not copy scsi_id to containerized directory")
+
+		// Create Disk
+		volName, volID := createAndValidateUniqueZonalDisk(client, p, z)
+
+		defer func() {
+			// Delete Disk
+			err := client.DeleteVolume(volID)
+			Expect(err).To(BeNil(), "DeleteVolume failed")
+
+			// Validate Disk Deleted
+			_, err = computeService.Disks.Get(p, z, volName).Do()
+			Expect(gce.IsGCEError(err, "notFound")).To(BeTrue(), "Expected disk to not be found")
+		}()
+
+		// Attach Disk
+		err = client.ControllerPublishVolume(volID, instance.GetNodeID())
+		Expect(err).To(BeNil(), "ControllerPublishVolume failed with error for disk %v on node %v: %v", volID, instance.GetNodeID())
+
+		defer func() {
+			// Detach Disk
+			err = client.ControllerUnpublishVolume(volID, instance.GetNodeID())
+			if err != nil {
+				klog.Errorf("Failed to detach disk: %v", err)
+			}
+
+		}()
 
+		// MESS UP THE symlink
+
+		err = testutils.RmAll(instance, "/dev/disk/by-id")
```

Comment:
this only tests the "symlink not found" path, need to augment test to also switch symlinks around and test that as well.